src/Microsoft.DotNet.Build.CloudTestTasks/UploadToAzure.cs

```diff
@@ -42,6 +42,16 @@ public class UploadToAzure : AzureConnectionStringBuildTask, ICancelableTask
         /// </summary>
         public bool Overwrite { get; set; } = false;
 
+        /// <summary>
+        /// Enables idempotency when Overwrite is false.
+        /// 
+        /// false: (default) Attempting to upload an item that already exists fails.
+        /// 
+        /// true: When an item already exists, download the existing blob to check if it's
+        /// byte-for-byte identical to the one being uploaded. If so, pass. If not, fail.
+        /// </summary>
+        public bool PassIfExistingItemIdentical { get; set; }
```

Comment:
Any desire to just make this the default behavior?

Reply:
No I think that is a bad idea to make this the default behavior. I fully expect this to negatively impact performance so we should make folks that need it opt-in. I do think we might want to consider other names for the property though, not coming up with a good suggestion right now though.

Reply:
Even on a functional level, I think it should remain off: it would be a bug in normal cases since we expect everything to have unique versions.